File: lavcode-node/src/actions/folder/dtos/create-folder.dto.ts
import { DtoRequired } from "@ipare/swagger";
import { IsNotEmpty, IsString } from "class-validator";
import { UpsertIconDto } from "../../../dtos/upsert-icon.dto";

export class CreateFolderDto {
  @DtoRequired()
  @IsString()
  name!: string;

  @DtoRequired()
  @IsNotEmpty()
  icon!: UpsertIconDto;
}

File: src/Lavcode.Service.Api/Dtos/CreateFolderDto.cs
using Newtonsoft.Json;

namespace Lavcode.Service.Api.Dtos
{
    public class CreateFolderDto
    {
        [JsonProperty("name")]
        public string Name { get; set; }

        [JsonProperty("icon")]
        public UpsertIconDto Icon { get; set; }
    }
}
